quantum/os_detection.c

```diff
@@ -175,6 +175,7 @@ void slave_update_detected_host_os(os_variant_t os) {
     detected_os = os;
     last_time   = timer_read_fast();
     debouncing  = true;
+    process_detected_host_os_kb(os);
```

Comment:
While i cant say i have any domain knowledge of this code, the change doesnt look correct to me.

`os_detection_task` would be running on the slave half. This is pushing state from potentially a somewhat time sensitive context so running user code could completely throw off timings ( or I want to say a background thread so the execution context would be wrong?). 

```c
void os_detection_task(void) {
    if (current_usb_device_state == USB_DEVICE_STATE_CONFIGURED) {
```

My expectation would be that the above check is updated to have some form of `is_keyboard_master()` check.

Reply:
Also, it might be better to call it from `slave_update_detected_host_os`, as this is the function that the transaction manager calls for the slave side code.  That way, it's only called when the state is changed.

Reply:
> os_detection_task would be running on the slave half

`os_detection_task` waits until USB is configured, from my understanding on the slave side USB is never used so it's essentially a noop.

I followed @drashna 's train of thought here to add this here.

Reply:
> os_detection_task waits until USB is configured, from my understanding on the slave side USB is never used so it's essentially a noop.

Which is why i mention it should be changed.

> I followed @drashna 's train of thought here to add this here.

As I initially mention, its potentially going to cause issues and should be implemented differently to extending `slave_update_detected_host_os`.